crates/oxc_linter/src/rules/react/style_prop_object.rs

```diff
@@ -0,0 +1,428 @@
+use oxc_ast::{
+    ast::{
+        Argument, Expression, JSXAttribute, JSXAttributeItem, JSXAttributeName, JSXAttributeValue,
+        JSXElementName, ObjectPropertyKind,
+    },
+    AstKind,
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{
+    ast_util::get_declaration_of_variable,
+    context::{ContextHost, LintContext},
+    rule::Rule,
+    utils::is_create_element_call,
+    AstNode,
+};
+
+fn style_prop_object_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Style prop value must be an object")
+        .with_help("Make sure the 'style' prop value is an object")
```

Comment:
This help message is redundant and adds no extra information that the diagnostic message does not already provide. Consider removing it or providing more information (e.g. what type we think `style` is, if not an object)